snuba/datasets/schemas/__init__.py

```diff
@@ -1,9 +1,29 @@
+from __future__ import annotations
+
 from abc import ABC, abstractmethod
 from typing import Mapping, List
 
 from snuba.clickhouse.columns import ColumnSet
 
 
+class RelationalSource(ABC):
```

Comment:
I called this RelationalSource because it makes our life a lot easier to make Query represent an abstract view of a Snuba query in relational terms (not Clickhouse specific). Building a totally abstract Query (completely abstract from the relational idea) does not seem very useful. We would still have to define the data access paradigm somehow in that Query class, so the choice of a relational query is arbitrary.

Reply:
Moreover, this is a first step into splitting the abstract data model from the Clickhouse specifics. This PR moves the abstract data model definition (column + relations) into its class. Who needs to deal with the data model (like query) deals with RelationalSource and does not have access to the full schema.